File: CS214/projects/proj01/RectangleArea.java
/* RectangleArea.java computes the area of a rectangle.
 *
 * Input: The width and height of the rectangle.
 * Precondition: The width and height are positive numbers.
 * Output: The area of the rectangle.
 *
 * Begun by: Dr. Adams, for CS 214, at Calvin College.
 * Completed by: Ryan Vreeke
 * Date: 12 Feb, 2020
 **********************************************************/


import java.util.Scanner;  // include class for Scanner

public class RectangleArea {

     /* function rectangleArea() computes a rectangle's area, given its width and height.
      * Parameter: width and height, double's
      * Precondition: width and height are not negative.
      * Returns: the area of the rectangle
      */
     public static double rectangleArea(double width, double height) {
        return width * height;               // return an expression
     } // rectangleArea method
	
  // main program
  public static void main(String[] args) {
      // prompt for width
      System.out.println("\n\nTo compute the area of a rectangle,");
      System.out.print(" enter its width: ");

      // Create a connection named keyboard to standard in
      Scanner keyboard = new Scanner(System.in);

      // Get the width from the user
      double width = keyboard.nextDouble();

      // prompt for height
      System.out.print(" enter its height: ");

      // Get the height from the user
      double height = keyboard.nextDouble();

      // output area
      System.out.println("\nThe area is " + rectangleArea(width, height) + "\n");
      System.out.printf("The area is %f\n\n", rectangleArea(width, height));
      System.out.printf("The area is %.15f\n\n", rectangleArea(width, height));
  } // main method

} // class RectangleArea

File: CS214/projects/proj01/rectangle_area.rb
#! /usr/bin/ruby
# rectangle_area.rb computes the area of a rectangle given its width and height
# 
# Input: the width and height of a rectangle
# Precondition: the width and height are not negative
# Output: the area of the rectangle.
#
# Begun by: Prof. Adams, for CS 214 at Calvin College.
# Completed by: Ryan Vreeke
# Date: 12 Feb, 2020
###############################################################

# function rectangleArea returns a rectangle's area, given its width and height
# Parameters: width and height, numbers
# Precondition: width > 0, height > 0.
# Returns: the area of a rectangle whose width is width, and height is height.
def rectangleArea(width, height)
    width * height
end

if __FILE__ == $0
   puts "To compute the area of a rectangle,"
   print " enter its width: "
   width = gets.chomp.to_f # asks for a width input value from the user
   print " enter its height: "
   height = gets.chomp.to_f # asks for a height input value from the user
   print "The rectangle's area is: "
   puts rectangleArea(width, height) # calls the rectangleArea function with the inputed width and height and prints the value
end